general/workspace.hpp

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) 2010-2023, Lawrence Livermore National Security, LLC. Produced
+// at the Lawrence Livermore National Laboratory. All Rights reserved. See files
+// LICENSE and NOTICE for details. LLNL-CODE-806117.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability visit https://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the BSD-3 license. We welcome feedback and contributions, see file
+// CONTRIBUTING.md for details.
+
+#ifndef MFEM_WORKSPACE_HPP
+#define MFEM_WORKSPACE_HPP
+
+#include "../linalg/vector.hpp"
+#include <forward_list>
+
+namespace mfem
+{
+
+// Forward declaration of internal::WorkspaceChunk
+namespace internal { class WorkspaceChunk; }
+
+/// @brief A vector used as a short-lived workspace for temporary calculations,
+/// created with Workspace::NewVector.
+///
+/// A WorkspaceVector is created using the Workspace bump allocator. The
+/// allocator can quickly and cheaply create new vectors, so that these vectors
+/// can be created and destroyed in loops without incurring the memory
+/// allocation and deallocation overhead.
+///
+/// WorkspaceVector%s should be used only for short-lived temporary storage;
+/// they are not intended to be stored in other classes.
+class WorkspaceVector : public Vector
```

Comment:
Just from a style point of view, I think newlines between the various declarations in the class are easier to read and also more consistent with the rest of the library. I don't think the space savings adds much here, and it's nicer with whitespace.

Reply:
Done in 399370db07a7ccd9ef53900307067ed56e4fa396.